tests/test_sft_trainer.py

```diff
@@ -808,3 +809,62 @@ def test_pretokenized_dataset_wrong_format():
         # is essentially swallowing a KeyError here.
         with pytest.raises(ValueError):
             sft_trainer.train(MODEL_ARGS, data_args, train_args, PEFT_PT_ARGS)
+
+
+def test_set_log_level_for_logger_default():
+    """
+    Ensure that the correct log level is being set
+    for python native logger and transformers logger
+    """
+
+    # Set env var TRANSFORMERS_VERBOSITY as None and test
+    os.unsetenv("TRANSFORMERS_VERBOSITY")
+    os.unsetenv("LOG_LEVEL")
+    os.unsetenv("SFT_TRAINER_CONFIG_JSON_ENV_VAR")
+    os.unsetenv("SFT_TRAINER_CONFIG_JSON_PATH")
+    train_args = copy.deepcopy(TRAIN_ARGS)
+
+    # TEST IF NO ENV VAR ARE SET AND NO CLI ARGUMENT IS PASSED
+    training_args, logger = sft_trainer.set_log_level(train_args)
+    assert logger.getEffectiveLevel() == logging.WARNING
+    assert training_args.log_level in ["passive", "warning"]
```

Comment:
Since no CLI flag is passed, `training_args.log_level == "passive"` correct? Since this is the default value set

Reply:
Yea, that should be the correct one. But, `train_args = copy.deepcopy(TRAIN_ARGS)`, `train_args` takes in default value of log_level as "warning", which is quite strange and couldn't figure out why.

Reply:
That is strange, doesn't look like its coming from transformers side and we set log_level to passive as well...
```py
from transformers import TrainingArguments
training_args = TrainingArguments(learning_rate=2e-5,per_device_train_batch_size=4, num_train_epochs=2, save_strategy="epoch", output_dir="output_dir")
print(training_args.log_level)
# 'passive'

```

Reply:
Yes only in unit test case, it basically takes default value as "warning". Attaching, the test fail logs. I will dig more into it.

<img width="790" alt="image" src="https://github.com/user-attachments/assets/042c7392-76ce-4d8e-8def-2e9991e985b2">

Reply:
With a thought in mind that above test cases must be tampering with TRAIN_ARGS and hence moved all logging unit tests at the start in the same file. And now this issue is fixed. Quite strange!!

Reply:
Yes I think it's because you are editing the training_args env var when you run set_log_level() so the tests are conflicting with each other. Because if I parse the flags standalone, the default is correctly set to "passive"
```py
from tuning.config import configs
TRAIN_ARGS = configs.TrainingArguments(
    num_train_epochs=5,
    per_device_train_batch_size=4,
    per_device_eval_batch_size=4,
    gradient_accumulation_steps=4,
    learning_rate=0.00001,
    weight_decay=0,
    warmup_ratio=0.03,
    lr_scheduler_type="cosine",
    logging_steps=1,
    include_tokens_per_second=True,
    packing=False,
    max_seq_length=4096,
    save_strategy="epoch",
    output_dir="tmp",
)
print(TRAIN_ARGS.log_level)
# 'passive'
```

Reply:
But if you are using a new TRAIN_ARGS....then they shouldn't conflict with each other. But perhaps the env vars were being leftover set. When I run the unit test with:
```
with mock.patch.dict(os.environ, {}, clear=True):
        print(TRAIN_ARGS.log_level)
        training_args, logger = sft_trainer.set_log_level(TRAIN_ARGS)
        assert logger.getEffectiveLevel() == logging.WARNING
        assert training_args.log_level == "passive"
```
it passes

Reply:
Got it!